tests/integration/test_runner_manager_openstack.py

```diff
@@ -0,0 +1,411 @@
+    runner_manager.create_runners(1)
+    runner_list = runner_manager.get_runners()
+    try:
+        await assert_runner_amount(runner_manager, 1)
+    except TimeoutError as err:
+        raise AssertionError("Test arrange failed: Expect one runner") from err
+
+    runner = runner_list[0]
+    assert (
+        runner.cloud_state == CloudRunnerState.ACTIVE
+    ), "Test arrange failed: Expect runner in active state"
+    try:
+        await wait_for(
+            lambda: runner_manager.get_runners()[0].github_state == GitHubRunnerState.IDLE,
+            timeout=120,
+            check_interval=10,
+        )
+    except TimeoutError as err:
+        raise AssertionError("Test arrange failed: Expect runner in idle state") from err
+    return runner_manager
+
+
+def workflow_is_status(workflow: Workflow, status: str) -> bool:
+    """Check if workflow in provided status.
+
+    Args:
+        workflow: The workflow to check.
+        status: The status to check for.
+
+    Returns:
+        Whether the workflow is in the status.
+    """
+    workflow.update()
+    return workflow.status == status
+
+
+async def assert_runner_amount(runner_manager: RunnerManager, num: int):
+    """Assert the number of runner a runner manager has.
+
+    A TimeoutError will be thrown if runners are still found after timeout.
+
+    Args:
+        runner_manager: The RunnerManager to check.
+        num: Number of runner to check for.
+    """
+    runner_list = runner_manager.get_runners()
+    assert isinstance(runner_list, tuple)
+    if len(runner_list) == num:
+        return
+
+    # The openstack server can take sometime to fully clean up or create.
+    await wait_for(lambda: len(runner_manager.get_runners()) == num)
+
+
+@pytest.mark.openstack
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_get_no_runner(runner_manager: RunnerManager) -> None:
+    """
+    Arrange: RunnerManager instance with no runners.
+    Act: Get runners.
+    Assert: Empty tuple returned.
+    """
+    runner_list = runner_manager.get_runners()
+    assert isinstance(runner_list, tuple)
+    assert not runner_list
+
+
+@pytest.mark.openstack
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_runner_normal_idle_lifecycle(
+    runner_manager: RunnerManager, openstack_runner_manager: OpenstackRunnerManager
+) -> None:
+    """
+    Arrange: RunnerManager instance with no runners.
+    Act:
+        1. Create one runner.
+        2. Run health check on the runner.
+        3. Delete all idle runner.
+    Assert:
+        1. An active idle runner.
+        2. Health check passes.
+        3. No runners.
+    """
+    # 1.
+    runner_id_list = runner_manager.create_runners(1)
+    assert isinstance(runner_id_list, tuple)
+    assert len(runner_id_list) == 1
+    runner_id = runner_id_list[0]
+
+    try:
+        await assert_runner_amount(runner_manager, 1)
```

Comment:
Should this be called `wait_runner_amount`?

Reply:
Fixed